[Flaking Test] UT TestComputePodActions*

### Which jobs are flaking?

https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-unit/2079717425376399360
https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-unit/2079686207066869760

### Which tests are flaking?

- TestComputePodActionsWithInitContainers 
- TestComputePodActionsWithContainerRestartRules

### Since when has it been flaking?

today

### Testgrid link

https://testgrid.k8s.io/sig-release-master-blocking#ci-kubernetes-unit

### Reason for failure (if possible)

```
{Failed  === RUN   TestComputePodActionsWithContainerRestartRules/resize_of_a_running_non-sidecar_init_container
    kuberuntime_manager_test.go:1891: 
        	Error Trace:	/home/prow/go/src/k8s.io/kubernetes/pkg/kubelet/kuberuntime/kuberuntime_manager_test.go:1891
        	            				/home/prow/go/src/k8s.io/kubernetes/pkg/kubelet/kuberuntime/kuberuntime_manager_test.go:2226
        	Error:      	Not equal: 
        	            	expected: &kuberuntime.podActions{KillPod:false, CreateSandbox:false, SandboxID:"sandboxID", Attempt:0x0, InitContainersToStart:[]int(nil), ContainersToStart:[]int{}, ContainersToKill:map[container.ContainerID]kuberuntime.containerToKillInfo{}, EphemeralContainersToStart:[]int(nil), ContainersToUpdate:map[v1.ResourceName][]kuberuntime.containerToUpdateInfo{"cpu":[]kuberuntime.containerToUpdateInfo{kuberuntime.containerToUpdateInfo{container:(*v1.Container)(0xc0007c8008), kubeContainerID:container.ContainerID{Type:"", ID:"initid1"}, desiredContainerResources:kuberuntime.resourceRequirements{memoryLimit:0, memoryRequest:838860800, cpuLimit:0, cpuRequest:800}, currentContainerResources:(*kuberuntime.resourceRequirements)(0xc00039d2e0)}}, "memory":[]kuberuntime.containerToUpdateInfo{kuberuntime.containerToUpdateInfo{container:(*v1.Container)(0xc0007c8008), kubeContainerID:container.ContainerID{Type:"", ID:"initid1"}, desiredContainerResources:kuberuntime.resourceRequirements{memoryLimit:0, memoryRequest:838860800, cpuLimit:0, cpuRequest:800}, currentContainerResources:(*kuberuntime.resourceRequirements)(0xc00039d2c0)}}}, VolumesToUpsize:[]v1.Volume(nil), VolumesToDownsize:[]v1.Volume(nil), UpdatePodResources:false, ContainersToReset:[]kuberuntime.containerToRemoveInfo(nil), UpdatePodLevelResources:false}
        	            	actual  : &kuberuntime.podActions{KillPod:false, CreateSandbox:false, SandboxID:"sandboxID", Attempt:0x0, InitContainersToStart:[]int(nil), ContainersToStart:[]int{}, ContainersToKill:map[container.ContainerID]kuberuntime.containerToKillInfo{}, EphemeralContainersToStart:[]int(nil), ContainersToUpdate:map[v1.ResourceName][]kuberuntime.containerToUpdateInfo{"cpu":[]kuberuntime.containerToUpdateInfo{kuberuntime.containerToUpdateInfo{container:(*v1.Container)(0xc00044ba00), kubeContainerID:container.ContainerID{Type:"", ID:"initid1"}, desiredContainerResources:kuberuntime.resourceRequirements{memoryLimit:0, memoryRequest:838860800, cpuLimit:0, cpuRequest:800}, currentContainerResources:(*kuberuntime.resourceRequirements)(0xc00039da60)}}, "memory":[]kuberuntime.containerToUpdateInfo{kuberuntime.containerToUpdateInfo{container:(*v1.Container)(0xc00044ba00), kubeContainerID:container.ContainerID{Type:"", ID:"initid1"}, desiredContainerResources:kuberuntime.resourceRequirements{memoryLimit:0, memoryRequest:838860800, cpuLimit:0, cpuRequest:800}, currentContainerResources:(*kuberuntime.resourceRequirements)(0xc00039da60)}}}, VolumesToUpsize:[]v1.Volume(nil), VolumesToDownsize:[]v1.Volume(nil), UpdatePodResources:false, ContainersToReset:[]kuberuntime.containerToRemoveInfo(nil), UpdatePodLevelResources:false}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -80,5 +80,5 @@
        	            	      memoryLimit: (int64) 0,
        	            	-     memoryRequest: (int64) 419430400,
        	            	+     memoryRequest: (int64) 0,
        	            	      cpuLimit: (int64) 0,
        	            	-     cpuRequest: (int64) 400
        	            	+     cpuRequest: (int64) 0
        	            	     })
        	            	@@ -154,5 +154,5 @@
        	            	      memoryLimit: (int64) 0,
        	            	-     memoryRequest: (int64) 419430400,
        	            	+     memoryRequest: (int64) 0,
        	            	      cpuLimit: (int64) 0,
        	            	-     cpuRequest: (int64) 400
        	            	+     cpuRequest: (int64) 0
        	            	     })
        	Test:       	TestComputePodActionsWithContainerRestartRules/resize_of_a_running_non-sidecar_init_container
        	Messages:   	resize of a running non-sidecar init container
--- FAIL: TestComputePodActionsWithContainerRestartRules/resize_of_a_running_non-sidecar_init_container (0.00s)

=== RUN   TestComputePodActionsWithContainerRestartRules/no_init_containers_have_been_started;_start_the_first_one
    kuberuntime_manager_test.go:1891: 
        	Error Trace:	/home/prow/go/src/k8s.io/kubernetes/pkg/kubelet/kuberuntime/kuberuntime_manager_test.go:1891
        	            				/home/prow/go/src/k8s.io/kubernetes/pkg/kubelet/kuberuntime/kuberuntime_manager_test.go:2226
        	Error:      	Not equal: 
        	            	expected: &kuberuntime.podActions{KillPod:false, CreateSandbox:false, SandboxID:"sandboxID", Attempt:0x0, InitContainersToStart:[]int{0}, ContainersToStart:[]int{}, ContainersToKill:map[container.ContainerID]kuberuntime.containerToKillInfo{}, EphemeralContainersToStart:[]int(nil), ContainersToUpdate:map[v1.ResourceName][]kuberuntime.containerToUpdateInfo{}, VolumesToUpsize:[]v1.Volume(nil), VolumesToDownsize:[]v1.Volume(nil), UpdatePodResources:false, ContainersToReset:[]kuberuntime.containerToRemoveInfo(nil), UpdatePodLevelResources:false}
        	            	actual  : &kuberuntime.podActions{KillPod:false, CreateSandbox:false, SandboxID:"sandboxID", Attempt:0x0, InitContainersToStart:[]int{0}, ContainersToStart:[]int{}, ContainersToKill:map[container.ContainerID]kuberuntime.containerToKillInfo{}, EphemeralContainersToStart:[]int(nil), ContainersToUpdate:map[v1.ResourceName][]kuberuntime.containerToUpdateInfo{}, VolumesToUpsize:[]v1.Volume(nil), VolumesToDownsize:[]v1.Volume(nil), UpdatePodResources:true, ContainersToReset:[]kuberuntime.containerToRemoveInfo(nil), UpdatePodLevelResources:false}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -17,3 +17,3 @@
        	            	  VolumesToDownsize: ([]v1.Volume) <nil>,
        	            	- UpdatePodResources: (bool) false,
        	            	+ UpdatePodResources: (bool) true,
        	            	  ContainersToReset: ([]kuberuntime.containerToRemoveInfo) <nil>,
        	Test:       	TestComputePodActionsWithContainerRestartRules/no_init_containers_have_been_started;_start_the_first_one
        	Messages:   	no init containers have been started; start the first one
--- FAIL: TestComputePodActionsWithContainerRestartRules/no_init_containers_have_been_started;_start_the_first_one (0.00s)

=== RUN   TestComputePodActionsWithContainerRestartRules
--- FAIL: TestComputePodActionsWithContainerRestartRules (0.02s)
}
```

### Anything else we need to know?

_No response_

### Relevant SIG(s)

/sig node

--- a/pkg/kubelet/kuberuntime/kuberuntime_manager_test.go
+++ b/pkg/kubelet/kuberuntime/kuberuntime_manager_test.go
@@ -1892,10 +1892,6 @@ func verifyActions(t *testing.T, expected, actual *podActions, desc string) {
 }
 
 func TestComputePodActionsWithInitContainers(t *testing.T) {
-	tCtx := ktesting.Init(t)
-	_, _, m, err := createTestRuntimeManager(tCtx)
-	require.NoError(t, err)
-
 	cpu400m := resource.MustParse("400m")
 	memory400Mi := resource.MustParse("400Mi")
 	cpu800m := resource.MustParse("800m")
@@ -2201,6 +2197,9 @@ func TestComputePodActionsWithInitContainers(t *testing.T) {
 				featuregatetesting.SetFeatureGateEmulationVersionDuringTest(t, utilfeature.DefaultFeatureGate, version.MustParse("1.36"))
 				featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.InPlacePodVerticalScalingInitContainers, false)
 			}
+
+			_, _, m, err := createTestRuntimeManager(tCtx)
+			require.NoError(t, err)
 
 			pod, status := makeBasePodAndStatusWithInitContainers()
 
@@ -2850,8 +2849,6 @@ func TestComputePodActionsWithContainerRestartRules(t *testing.T) {
 		containerRestartPolicyOnFailure = v1.ContainerRestartPolicyOnFailure
 		containerRestartPolicyNever     = v1.ContainerRestartPolicyNever
 	)
-	_, _, m, err := createTestRuntimeManager(tCtx)
-	require.NoError(t, err)
 
 	// Creating a pair reference pod and status for the test cases to refer
 	// the specific fields.
@@ -2965,6 +2962,8 @@ func TestComputePodActionsWithContainerRestartRules(t *testing.T) {
 			},
 		},
 	} {
+		_, _, m, err := createTestRuntimeManager(tCtx)
+		require.NoError(t, err)
 		pod, status := makeBasePodAndStatus()
 		if test.mutatePodFn != nil {
 			test.mutatePodFn(pod)

